dll/win32/shell32/utils.cpp

```diff
@@ -103,6 +103,68 @@ SHInvokePrivilegedFunctionW(
     return hr;
 }
 
+/*************************************************************************
+ *                SHTestTokenPrivilegeW (SHELL32.236)
+ *
+ * @see http://undoc.airesoft.co.uk/shell32.dll/SHTestTokenPrivilegeW.php
+ */
+EXTERN_C
+BOOL WINAPI
+SHTestTokenPrivilegeW(_In_opt_ HANDLE hToken, _In_z_ LPCWSTR lpName)
+{
+    UINT iPriv, cPrivs;
+    LUID Luid, *pPrivLuid;
+    DWORD dwLength;
+    TOKEN_PRIVILEGES *pTokenPriv;
+    HANDLE hTokenChoice = hToken, hNewToken = NULL;
+    BOOL ret = FALSE;
+
+    TRACE("(%p, %s)\n", hToken, debugstr_w(lpName));
+
+    if (!lpName)
+        return FALSE;
+
+    if (!hToken)
+    {
+        if (!SHOpenEffectiveToken(&hNewToken))
+            goto Quit;
+
+        if (!hNewToken)
+            return FALSE;
+
+        hTokenChoice = hNewToken;
+    }
+
+    if (!LookupPrivilegeValueW(NULL, lpName, &Luid))
+        return FALSE;
+
+    dwLength = 0;
+    GetTokenInformation(hTokenChoice, TokenPrivileges, NULL, 0, &dwLength);
```

Comment:
Note that this call might fail for other reasons, and failure needs to be checked for.

Reply:
OK.